ping/ping_output.c

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (c) 2024 Georg Pfuetzenreuter <mail+ip@georg-pfuetzenreuter.net>
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ * 3. Neither the name of the copyright holder nor the names of its
+ *    contributors may be used to endorse or promote products derived from
+ *    this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "ping.h"
+#include "stdarg.h"
+
+void ping_print_int(struct ping_rts *rts, char *msg, char *json_key, int json_val) {
+	if (rts->opt_json) {
+		construct_json(rts, PING_JSON_INT, json_key, json_val);
+	} else {
+		printf(msg, json_val);
+	}
+
+}
+
+void ping_print_uint(struct ping_rts *rts, char *msg, char *json_key, unsigned int json_val) {
+	if (rts->opt_json) {
+		construct_json(rts, PING_JSON_UINT, json_key, json_val);
+	} else {
+		printf(msg, json_val);
+	}
+
+}
+
+void ping_print_str(struct ping_rts *rts, char *msg, char *json_key, char *json_val) {
+	if (rts->opt_json) {
+		construct_json(rts, PING_JSON_STR, json_key, json_val);
+	} else {
+		printf(msg, json_val);
+	}
+}
+
+inline void ping_print_version(struct ping_rts *rts) {
+	if (rts->opt_json) {
+		construct_json(rts, PING_JSON_STR, "version", PACKAGE_VERSION);
+		print_json_packet(rts);
+	} else {
+		printf(PACKAGE_VERSION);
+		print_config();
+	}
+}
+
+inline void ping_print_finish(struct ping_rts *rts) {
+	if(!rts->opt_json) {
+		putchar('\n');
+		fflush(stdout);
+	}
+}
+
+void ping_error(struct ping_rts *rts, int status, int errnum, char *msg) {
```

Comment:
This function is not needed, the following `ping_errorf()` can handle the case when only unformatted message is passed. Please delete this function and rename `ping_errorf()` to `ping_error()`.

Reply:
Thanks for the hint, adjusted.